File: cakk-api/src/main/java/com/cakk/api/dto/response/shop/CakeShopSearchResponse.java
package com.cakk.api.dto.response.shop;

import java.util.List;

import lombok.Builder;

import com.cakk.domain.mysql.dto.param.shop.CakeShopSearchResponseParam;

@Builder
public record CakeShopSearchResponse(
	List<CakeShopSearchResponseParam> cakeShops,
	Long lastCakeShopId,
	int size
) {
}

File: cakk-api/src/main/java/com/cakk/api/dto/response/shop/CakeShopSearchResponse.kt
package com.cakk.api.dto.response.shop

import com.fasterxml.jackson.annotation.JsonTypeInfo

import com.cakk.domain.mysql.dto.param.shop.CakeShopSearchResponseParam

@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
data class CakeShopSearchResponse(
	val cakeShops: List<CakeShopSearchResponseParam>,
	val lastCakeShopId: Long?,
	val size: Int
)
